### PR TITLE
feat(events): add scroll to top button

### PR DESCRIPTION
**Pull request by lex111**
Demo:

image

### CODE REVIEW

File: src/auth/screens/events.screen.js [hunk 1]
```diff
@@ -77,6 +82,14 @@ const styles = StyleSheet.create({
 });
 
 class Events extends Component {
+  constructor() {
+    super();
```
Comment (chinesedfan) [inline src/auth/screens/events.screen.js]: `super(props)` may be better.

File: src/components/back-to-top.component.js [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+  render() {
+    const { onPress, bottom } = this.props;
+    const extraStyles = {
+      bottom,
+      opacity: this.state.animatedOpacity,
+    };
+
+    return (
+      <Container style={extraStyles}>
+        <TouchableOpacity onPress={() => onPress()}>
```
Comment (chinesedfan) [inline src/components/back-to-top.component.js]: Can here be written as `onPress={onPress}` directly?

File: src/auth/screens/events.screen.js [hunk 3]
```diff
@@ -77,6 +82,14 @@ const styles = StyleSheet.create({
...
+
+    this.state = {
+      offsetY: null,
```
Comment (chinesedfan) [inline src/auth/screens/events.screen.js]: If `offset` is number, should it be zero by default?

File: src/components/back-to-top.component.js [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+          <Icon color={colors.white} size={32} name="expand-less" />
+        </TouchableOpacity>
+      </Container>
+    );
+  }
+}
+
+BackToTop.defaultProps = {
+  offsetY: null,
```
Comment (chinesedfan) [inline src/components/back-to-top.component.js]: Similar here.

File: src/components/back-to-top.component.js [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+  opacity: 1,
+  bottom: 10,
+  distance: 200,
+  onPress: undefined,
```
Comment (chinesedfan) [inline src/components/back-to-top.component.js]: The default value is `undefined`, is it not meaningful?

**Comment by ocarreterom** [comment]
This functionality would not be necessary in iOS devices, because touching the status bar, the view scrolls to top. 🤔 
What do you thing?

**Comment by andrewda** [comment]
Good point @ocarreterom. I still kinda like the button, but I guess either way works for iOS.

**Comment by machour** [comment]
So, do we keep this for iOS, or do we restrict it for Android only? 🤔

**Comment by andrewda** [comment]
@machour I still like it as is – keep the button on iOS.

**Comment by ocarreterom** [comment]
@andrewda in my opinion if you want quick action to scroll to top, touching the current active tab is the best improvement. Android and iOS have this feature in common. I don't use any app that use a floating button to scroll.

**Comment by housseindjirdeh** [comment]
Since we haven't come to a consensus here, going to close this for now (thank you thank you @lex111 for putting the work to make this work).

Let's keep the discussion flowing if somebody happens to think this would be nice to have and worthwhile. Thoughts @machour @chinesedfan?

**Comment by chinesedfan** [comment]
I came up with a compromised solution that only setting the button visible when users scroll to top.

**Comment by machour** [comment]
To be honest, I've never been a fan of this kind of button on websites, but I'm fine living with it if it's implemented (maybe a togglable configuration?)

**Pull request closed without merging**